icgcget/cli.py

```diff
@@ -76,103 +77,102 @@ def logger_setup(logfile):
-                                              config['cghub.transport.parallel'], args.output)
+        if manifest is True:
+            code = gt_client.genetorrent_manifest_call(fileid, config['cghub.access'], config['cghub.tool'],
+                                                       config['cghub.transport.parallel'], output)
+        elif fileid is not None:
+            code = gt_client.genetorrent_call(file, config['cghub.access'], config['cghub.tool'],
+                                              config['cghub.transport.parallel'], output)
         if code != 0:
-            logger.error(args.repo + " exited with a nonzero error code.")
+            logger.error(repo + " exited with a nonzero error code.")
 
-    elif args.repo == 'gdc':
-        if args.manifest is not None:
-            code = gdc_client.gdc_manifest_call(args.manifest, config['gdc.access'], config['gdc.tool'], args.output,
+    elif repo == 'gdc':
+        if manifest is True:
+            code = gdc_client.gdc_manifest_call(fileid, config['gdc.access'], config['gdc.tool'], output,
                                                 config['udt'], config['gdc.transport.parallel'])
-        if args.file is not None:
-            code = gdc_client.gdc_call(args.file, config['gdc.access'], config['gdc.tool'], args.output, config['udt'],
+        elif fileid is not None:
+            code = gdc_client.gdc_call(fileid, config['gdc.access'], config['gdc.tool'], output, config['udt'],
                                        config['gdc.transport.parallel'])
         if code != 0:
-            logger.error(args.repo + " exited with a nonzero error code.")
+            logger.error(repo + " exited with a nonzero error code.")
 
     return code
 
-def main():
-    repos = ['ega', 'collab', 'cghub', 'aws', 'gdc']
-    parser = argparse.ArgumentParser()
-    parser.add_argument("--config", nargs='?', default="/icgc/conf/config.yaml",
-                        help="File used to set download preferences and authentication criteria")
-    parser.add_argument('repo', choices=repos, help='Specify which repository to download from, all lowercase letters')
-    parser.add_argument('-f', '--file', nargs='*', help='Lowercase identifier of file or path to manifest file')
-    parser.add_argument('-m', '--manifest', help='Flag used when the downloading from a manifest file')
-    parser.add_argument('--output', nargs='?', default='/icgc/mnt/downloads', help='Directory to save downloaded files')
-    parsed_args = parser.parse_args()
-    rc = call_client(parsed_args)
-    sys.exit(rc)
+
+@click.command()
+@click.option('--config', type=click.File(), default="/icgc/conf/config.yaml")
+@click.argument('repo', type=click.Choice(['ega', 'collab', 'cghub', 'aws', 'gdc']))
+@click.argument('fileid', nargs=-1)
+@click.option('--manifest/--no-manifest', ' /-m', default="false")
+@click.option('--output', default='/icgc/mnt/downloads', type=click.Path(exists=False))
```

Comment:
I don't know if that should be the case in the tool proper. Sounds like more of a Docker environment behavior that we may want to remove from the main logic.